src/main/java/dev/aziz/grocerystore/services/BasketItemService.java

```diff
@@ -72,8 +78,13 @@ public BasketTotalDto getItems(String login) {
         List<BasketItem> basketByUser = basketItemRepository.findBasketsByUserId(user.getId())
                 .orElseThrow(() -> new AppException("Basket not found.", HttpStatus.NOT_FOUND));
 
+        List<UserPromotion> userPromotionList = userPromotionRepository.findUserPromotionsByUserId(user.getId());
+
+        List<PromotionConfig> promotions = promotionConfigRepository.findAll();
```

Comment:
You don't need to know all the available promotions, the user's promotions is enough